src/Http/RequestFactory.php

```diff
@@ -60,25 +60,59 @@ public function setProxy($proxy)
 	 */
 	public function createHttpRequest()
 	{
+		$remoteAddr = isset($_SERVER['REMOTE_ADDR']) ? $_SERVER['REMOTE_ADDR'] : NULL;
+		$remoteHost = isset($_SERVER['REMOTE_HOST']) ? $_SERVER['REMOTE_HOST'] : NULL;
+
+		// use real client address and host if trusted proxy is used
+		$usingTrustedProxy = FALSE;
+		foreach ($this->proxies as $proxy) {
+			if (Helpers::ipMatch($remoteAddr, $proxy)) {
+				$usingTrustedProxy = TRUE;
+				break;
+			}
+		}
+
+		if ($usingTrustedProxy) {
+			if (isset($_SERVER['HTTP_X_FORWARDED_FOR'])) {
+				$remoteAddr = trim(current(explode(',', $_SERVER['HTTP_X_FORWARDED_FOR'])));
+			}
+
+			if (isset($_SERVER['HTTP_X_FORWARDED_HOST'])) {
+				$remoteHost = trim(current(explode(',', $_SERVER['HTTP_X_FORWARDED_HOST'])));
+			}
+		}
```

Comment:
Shouldn't this be grouped together with this code (it is about 20 lines below)? I don't see anything preventing it.

```
`if ($usingTrustedProxy && !empty($_SERVER['HTTP_X_FORWARDED_PORT'])) {
	$port = (int) $_SERVER['HTTP_X_FORWARDED_PORT'];
}
```